src/foraging_gui/Foraging.py

```diff
@@ -1216,6 +1223,105 @@ def _GetSettings(self):
+                                          f'Waterlog needs to be added manually to Slims')
+                            QMessageBox.critical(self,
+                                                 'Adding Waterlog',
+                                                 f'Mouse {session.subject_id} exists in Slims but contains validations errors.'
+                                                 f'Waterlog needs to be added manually to Slims',
+                                                 QMessageBox.Ok)
+                            return
+                        else:
+                            raise e
+
+                else:   # Not enough info in schedule to create mouse
+                    logging.error(f'Mouse {session.subject_id} does not exist in Slims, and schedule does not contain '
+                                  f'enough information to add mouse. Waterlog needs to be added manually to Slims')
+                    QMessageBox.critical(self,
+                                         'Adding Waterlog',
+                                         f'Mouse {session.subject_id} does not exist in Slims, and schedule does not contain '
+                                         f'enough information to add mouse. Waterlog needs to be added manually to Slims',
+                                         QMessageBox.Ok)
+                    return
+            else:
+                raise e
+
+        # extract water information
+        water_json = session.stimulus_epochs[0].output_parameters.water.items()
+        water = {k: v if not (isinstance(v, float) and math.isnan(v)) else None for k, v in water_json}
+
+        # extract software information
+        software = session.data_streams[0].software[0]
+
+        # create model
+        model = models.SlimsWaterlogResult(
+            mouse_pk=mouse.pk,
+            date=session.session_start_time,
+            weight_g=session.animal_weight_prior,
+            water_earned_ml=water['water_in_session_foraging'],
+            water_supplement_delivered_ml=water['water_after_session'],
+            water_supplement_recommended_ml=None,
+            total_water_ml=water['water_in_session_total'],
+            comments=session.notes,
+            workstation=session.rig_id,
+            sw_source=software.url,
+            sw_version=software.version,
+            test_pk=self.slims_client.fetch_pk("Test", test_name="test_waterlog"))
+
+        # check if mouse already has waterlog for at session time and if, so update model
+        waterlogs = self.slims_client.fetch_models(models.SlimsWaterlogResult, mouse_pk=mouse.pk)
+        if waterlogs != [] and waterlogs[0].date.strftime("%Y-%m-%d %H:%M:%S") == \
+                session.session_start_time.astimezone(timezone.utc).strftime("%Y-%m-%d %H:%M:%S"):
+            model.pk = waterlogs[0].pk
+            self.slims_client.update_model(model=model)
+        else:
+            self.slims_client.add_model(model)
```

Comment:
just add more logging.info statements through here to help with future debugging